### DISCUSSION TITLE
CoreDNS serve non-clustered zones

### DISCUSSION THREAD

**Discussion by orzen** [k3s-io/k3s#4000] [Q&A] [answered]
I just started to look into k3s and Kubernetes in general so I apologize ahead of time for lacking terminology and understanding of both Kubernetes and k3s.

I currently hosting two DNS name servers(currently hosted on VPS) and I was thinking about killing two birds with one stone and replace one name server with a cluster-hosted instance of CoreDNS. Since k3s already ships with a CoreDNS instance I was wondering if it's possible to combine the host of "external" zones (for services hosted outside of the cluster) and zones mapped to services in the cluster?

1. Is it possible to arrange this with CoreDNS that is "shipped" (kube-system) with k3s?
2. Is it bad practice to use the same instance for mixed purposes?
3. Does it open up for vulnerabilities?

Thank you in advance!

**Comment by brandond** [comment]
coredns is only really intended to be used within a Kubernetes cluster. It is not a full-featured DNS server intended for general-purpose use.

**Reply by orzen** [reply]
Thank you for your reply. My impression is that CoreDNS is a general-purpose DNS server.

**Comment by orzen** [comment, marked as answer]
Someone asked a similar question in the #coredns at Slack.

**Question:**
> Hi folks! :wave:
> We´re looking to replace all our DNS with CoreDNS on Kubernetes. External AND Internal.
> For a rather big enterprise company, one the biggest E-commerce of the Nordics with loads of traffic.
> For dealing with overloads and avoid failures, we´re looking to host AKS clusters ONLY having CoreDNS in it, in two different regions - Is this a good idea?
> Im thinking re-using some ConfigMap and deploy to different regional clusters, and maybe some Plugin to CoreDNS that syncs DNS stuff.
> Is my thinking correct? Looking forward for input from you pro's. :slightly_smiling_face:

**Answer:**
> Cluster internally it most likely does already run CoreDNS, but that depends on the k8s version.
> On the external side. Do you mean external records for internal queries? That could be done via the cluster one and ExternalService.
> Additionally for customer facing DNS queries you would run your own CoreDNS instances as a general service separate from the cluster one. Could also be helpful for more specialized queries within your cluster to have a bit more control over caching or query rewriting.
> There are a lot of variables and missing context to get into more details. Happy to chat about it

**Reply by brandond** [reply]
Originally it didn't implement zone transfers, but it looks like they've added that, even if they don't behave properly as a secondary. I guess I personally wouldn't chose it to host a public zone, over a more traditional package, but that might just be personal bias at this point.
